packages/serverpod/lib/src/server/log_manager/log_manager.dart

```diff
@@ -271,6 +274,8 @@ class SessionLogManager {
     required Session session,
   }) async {
     await _openStreamLogLock.synchronized(() async {
+      if (_isLoggingOpened) return;
+
```

Comment:
Suggestion: Remove internal check.

Since we guard against this on the callsite on line 251 this should not be required right?

Reply:
The internal check is vital as `n` number of methods can acquire the lock but only the first should execute the init logic. So the other methods have to eject if the variable was set.